### PR TITLE
Send hostname and pid for identification

### PR DESCRIPTION
**Pull request by szdavid92**

### CODE REVIEW

File: lib/agent/api/index.spec.js [hunk 1]
```diff
@@ -41,7 +41,8 @@ describe('The Trace CollectorApi module', function () {
     collectorApi.sendRpmMetrics(data)
 
     expect(sendStub).to.have.been.calledOnce
-    expect(sendStub).to.have.been.calledWith(sendUrl, data)
+    expect(sendStub.args[0][0]).to.eql(sendUrl)
```
Comment (gergelyke) [inline lib/agent/api/index.spec.js]: how about `sendStub.to.be.calledWith()`?

File: lib/utils/configReader.spec.js [hunk 2]
```diff
@@ -236,20 +264,19 @@ describe('Config Reader module', function () {
...
     try {
       var config = configReader.getConfig()
-      expect(config.ignoreHeaders).to.eql(ignoreHeadersOption)
+      expect(config.ignoreHeaders).to.eql(wellformed)
       expect(log).to.have.been.calledOnce
+    } finally {
       delete process.env.TRACE_IGNORE_HEADERS
-    } catch (err) {
-      delete process.env.TRACE_IGNORE_HEADERS
-      throw err
```
Comment (gergelyke) [inline lib/utils/configReader.spec.js]: why have you deleted it? It will pass if an error is thrown

File: lib/utils/configReader.spec.js [hunk 3]
```diff
@@ -116,7 +124,27 @@ describe('Config Reader module', function () {
...
+
+    this.sandbox.stub(configReader, '_getSystemConfig', function () {
+      return { test: 'system' }
+    })
+
+    this.sandbox.stub(configReader, '_getEnvVarConfig', function () {
+      return { test: 'env' }
+    })
+
+    this.sandbox.stub(configReader, '_getFileConfig', function () {
```
Comment (gergelyke) [inline lib/utils/configReader.spec.js]: `this.sandbox.stub(configReader, '_getFileConfig').returns({})

File: lib/utils/configReader.spec.js (hunk 2 above)
Comment (szdavid92) [inline lib/utils/configReader.spec.js]: i revert this, it's broken, yes

File: lib/agent/api/index.spec.js (hunk 1 above)
Comment (szdavid92) [inline lib/agent/api/index.spec.js]: how could i assert the subset next line?

Comment (gergelyke) [inline lib/agent/api/index.spec.js]: don't expect for a subset, expect for the whole thing

**Pull request merged**